aspnetcore/mvc/advanced/custom-formatters/Sample/Formatters/VcardOutputFormatter.cs

```diff
@@ -0,0 +1,76 @@
+﻿using Microsoft.AspNetCore.Mvc.Formatters;
+using Microsoft.AspNetCore.Http;
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using System.Text;
+using CustomFormatterDemo.Models;
+using Microsoft.Net.Http.Headers;
+using System.Reflection;
+using Microsoft.Extensions.Logging;
+
+namespace CustomFormatterDemo.Formatters
+{
+    #region classdef
+    public class VcardOutputFormatter : TextOutputFormatter
+    #endregion
+    {
+        #region ctor
+        public VcardOutputFormatter()
+        {
+            SupportedMediaTypes.Add(MediaTypeHeaderValue.Parse("text/vcard"));
+
+            SupportedEncodings.Add(Encoding.UTF8);
+            SupportedEncodings.Add(Encoding.Unicode);
+        }
+        #endregion
+
+        #region canwritetype
+        protected override bool CanWriteType(Type type)
+        {
+            if (typeof(Contact).IsAssignableFrom(type) 
```

Comment:
For this case, let keep simple and just check if the `type == typeof(Contact)`, I say this because if we were to correctly implement this whole thing, we should also handle the case where an action returns for example `IEnumerable<VipContact>` (where VipContact derives from Contact) and the other check for enumerable would not work in this case.